vault/expiration.go

```diff
@@ -114,6 +119,91 @@ func (c *Core) stopExpiration() error {
 	return nil
 }
 
+// Tidy cleans up stale leases which are associated with invalid tokens
+func (m *ExpirationManager) Tidy() error {
+	var tidyErrors *multierror.Error
+
+	// Create a cache to keep track of looked up tokens
+	tokenCache := make(map[string]string)
+
+	tidyFunc := func(leaseID string) {
+		m.logger.Trace("expiration: checking if lease is valid", "lease_id", leaseID)
```

Comment:
This is likely to be way too verbose, since this will print out for every single lease -- imagine if you have 200k of them. What would be better is a log line before this happens, then a running count, e.g. if the current count is divisible evenly by 500, print out an update line.